src/components/Editor/index.js

```diff
@@ -102,10 +103,17 @@ class Editor extends Component {
     if (this.props.onKeyDown) {
       this.props.onKeyDown(evt)
     }
-    if (evt.keyCode === 9 && !this.props.ignoreTabKey) { // Tab Key
+    
+    if (evt.keyCode === 9 && !this.state.tabGuarded) { // Tab Key
       document.execCommand('insertHTML', false, '&#009')
       evt.preventDefault()
-    } else if (evt.keyCode === 13) { // Enter Key
+    }else if (evt.keyCode === 27) {// Esc Key
+      this.setState({tabGuarded:true})
+    }else if(!(evt.shiftKey || evt.ctrlKey || evt.altKey)){
+      this.setState({tabGuarded:false})
```

Comment:
Hm, okay. I see what prompted you to use the flag before. This state change triggers early so that the selection save (`this.selection = ...`) in the keyup handler isn't respected on enter.

So now when the enter key is pressed the state changes, the component updates, and the selection is harmed.

We can try an easy fix. First of all we can add the flags to the if statement `&& this.state.tabGuarded) {` etc

Then we can maybe move this logic into the keyup handler. It has to be after the rest of the logic there. ^^

Makes sense?